components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/AccountLockHandler.java

```diff
@@ -492,16 +506,20 @@ protected boolean handlePostSetUserClaimValues(Event event, String userName, Use
                 if (notificationInternallyManage) {
                     if (StringUtils.isNotEmpty(existingAccountStateClaimValue)) {
                         // Send locked email only if the accountState claim value is PENDIG_SR or PENDING_EV.
-                        if (!existingAccountStateClaimValue.equals(AccountConstants.PENDING_SELF_REGISTRATION) && !existingAccountStateClaimValue.equals(AccountConstants.PENDING_EMAIL_VERIFICATION)) {
-                            triggerNotification(event, userName, userStoreManager, userStoreDomainName, tenantDomain, identityProperties,
-                                    AccountConstants.EMAIL_TEMPLATE_TYPE_ACC_LOCKED);
+                        if (isAdminInitiated) {
+                            triggerNotification(event, userName, userStoreManager, userStoreDomainName, tenantDomain,
+                                    identityProperties,
+                                    AccountConstants.EMAIL_TEMPLATE_TYPE_ACC_LOCKED_ADMIN_TRIGGERED);
+                        } else {
+                            triggerNotification(event, userName, userStoreManager, userStoreDomainName, tenantDomain,
+                                    identityProperties, AccountConstants.EMAIL_TEMPLATE_TYPE_ACC_LOCKED_FAILED_ATTEMPT);
+                        }
+                        if (!existingAccountStateClaimValue.equals(AccountConstants.PENDING_SELF_REGISTRATION)
```

Comment:
Swap constant with the variable to avoid possible null pointers.

Reply:
done